packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -279,11 +279,20 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
                 return;
               }
 
-              NSDictionary *jsError = RCTJSErrorFromCodeMessageAndNSError(code, message, error);
-              reject->call([jsError](jsi::Runtime &rt, jsi::Function &jsFunction) {
-                jsFunction.call(rt, convertObjCObjectToJSIValue(rt, jsError));
+              NSDictionary *jsErrorDetails = RCTJSErrorFromCodeMessageAndNSError(code, message, error);
+              reject->call([jsErrorDetails](jsi::Runtime &rt, jsi::Function &jsFunction) {
+                // From JS documentation:
+                // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#cause an
+                // Error can be created with `new Error(message, option);`. the `option` param is a JS object with the
+                // `cause` property. Create a valid `option` object
+                NSDictionary<NSString *, id> *jsErrorOptions = @{@"cause" : jsErrorDetails};
+                auto jsiObjCError = convertObjCObjectToJSIValue(rt, jsErrorOptions);
```

Comment:
Create a `jsi::Object` directly here and set the cause key directly, rather than needlessly going through objective-c.
```suggestion
                jsi::Object jsErrorOptions = jsi::Object(rt);
                jsErrorOptions.setProperty(rt, "cause", convertObjCObjectToJSIValue(rt, jsErrorOptions));
```